src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2549,6 +2560,100 @@ private function registerUidConfiguration(array $config, ContainerBuilder $conta
+            }
+
+            foreach ($sanitizerConfig['drop_elements'] ?? [] as $element) {
+                $def->addMethodCall('dropElement', [$element], true);
+            }
+
+            // Configures attributes
+            foreach ($sanitizerConfig['allow_attributes'] ?? [] as $attribute => $elements) {
+                $def->addMethodCall('allowAttribute', [$attribute, $elements], true);
+            }
+
+            foreach ($sanitizerConfig['drop_attributes'] ?? [] as $attribute => $elements) {
+                $def->addMethodCall('dropAttribute', [$attribute, $elements], true);
+            }
+
+            // Force attributes
+            foreach ($sanitizerConfig['force_attributes'] ?? [] as $element => $attributes) {
+                foreach ($attributes as $attrName => $attrValue) {
+                    $def->addMethodCall('forceAttribute', [$element, $attrName, $attrValue], true);
+                }
+            }
+
+            // Settings
+            if (isset($sanitizerConfig['force_https_urls'])) {
+                $def->addMethodCall('forceHttpsUrls', [$sanitizerConfig['force_https_urls']], true);
+            }
+
+            if (isset($sanitizerConfig['allowed_link_schemes'])) {
+                $def->addMethodCall('allowLinkSchemes', [$sanitizerConfig['allowed_link_schemes']], true);
+            }
+
+            if (isset($sanitizerConfig['allowed_link_hosts'])) {
+                $def->addMethodCall('allowLinkHosts', [$sanitizerConfig['allowed_link_hosts']], true);
+            }
+
+            if (isset($sanitizerConfig['allow_relative_links'])) {
+                $def->addMethodCall('allowRelativeLinks', [$sanitizerConfig['allow_relative_links']], true);
+            }
+
+            if (isset($sanitizerConfig['allowed_media_schemes'])) {
+                $def->addMethodCall('allowMediaSchemes', [$sanitizerConfig['allowed_media_schemes']], true);
+            }
+
+            if (isset($sanitizerConfig['allowed_media_hosts'])) {
+                $def->addMethodCall('allowMediaHosts', [$sanitizerConfig['allowed_media_hosts']], true);
+            }
+
+            if (isset($sanitizerConfig['allow_relative_medias'])) {
+                $def->addMethodCall('allowRelativeMedias', [$sanitizerConfig['allow_relative_medias']], true);
+            }
+
+            // Custom attribute sanitizers
+            foreach ($sanitizerConfig['with_attribute_sanitizers'] ?? [] as $serviceName) {
+                $def->addMethodCall('withAttributeSanitizer', [new Reference($serviceName)], true);
+            }
+
+            foreach ($sanitizerConfig['without_attribute_sanitizers'] ?? [] as $serviceName) {
+                $def->addMethodCall('withoutAttributeSanitizer', [new Reference($serviceName)], true);
+            }
+
+            // Create the sanitizer and link its config
+            $container->register($sanitizerName, HtmlSanitizer::class)
+                ->addArgument(new Reference($sanitizerName.'.config'))
```

Comment:
Registering the config using a suffix on the user-defined name while the sanitizer service does not use any suffix makes FrameworkBundle fragile. If someone registers a sanitizer name ending in `.config`, it will override the config service of another sanitizer and break things.

Reply:
there's no way around such risks if we don't force a convention on the sanitizer name
but the risk is theoretical to me, nobody will use that conflicting name in practice

Reply:
Well, I know it happened in the past for another component (I don't remember which one as that was years ago), with someone reporting a very weird bug.